sympy/solvers/diophantine.py

```diff
@@ -875,9 +875,9 @@ def _diop_quadratic(var, coeff, t):
                     + (sqa*g*u**2 + D*u + sqa*F) // _c
 
                 for z0 in range(0, abs(_c)):
-                    if divisible(
-                            sqa*g*z0**2 + D*z0 + sqa*F,
-                            _c):
+                    _temp = sqa*g*z0**2 + D*z0 + sqa*F
+                    # Check if the values of y and x obtained are integers or not
+                    if divisible(_temp, _c) and divisible(_temp*e*sqc - z0*_c, sqa*_c):
```

Comment:
It should be needed to check that also x is integral. The existing check is only for y. So it is  a second check in addition to the existing one.